loki/transform/transform_hoist_variables.py

```diff
@@ -150,6 +151,8 @@ def transform_subroutine(self, routine, **kwargs):
         call_map = CaseInsensitiveDict((str(call.name), call) for call in calls)
 
         for child in successors:
+            if not isinstance(child, SubroutineItem):
```

Comment:
This is currently not covered in the test base, which is why this bug existed in the first place.
I would suggest to amend the existing test [`test_single_column_coalesced_hoist_openacc`](https://github.com/ecmwf-ifs/loki/blob/c86b83aea180c6823ba0615ce395dc7b8dd51e35/transformations/tests/test_single_column_coalesced.py#L711) by moving the integer `c` to a new module and importing it in the kernel. 

Here's an untested implementation suggestion:

```python
fcode_module = """
module my_scaling_value_mod
    implicit none
    REAL :: c = 5.345
end module my_scaling_value_mod
""".strip()

[...]

module_source = Sourcefile.from_source(fcode_module, frontend=frontend)
module = module_source['my_scaling_value_mod']

[...]

kernel.enrich(module) # do this before enriching the driver

[...]

module_item = GlobalVarImportItem(name='my_scaling_value_mod#c', source=module_source)

[...]

    # This replaces the current implementation
    for transform in scc_transform:
        transform.apply(driver, role='driver', item=driver_item, targets=['compute_column'], successors=[kernel_item])
        transform.apply(kernel, role='kernel', item=kernel_item), successors=[module_item])

```

Reply:
Thanks, this was a good idea. 

I changed the test to the best of my ability, please check to see it is OK. I had to add the successors argument to the analysis part (a bit later than you suggested). I also ran the modified test in the main branch to make sure that it fails there.